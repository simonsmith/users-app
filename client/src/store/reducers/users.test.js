import usersReducer from './users';

test('should handle no state being passed', () => {
  expect(usersReducer(undefined, {})).toMatchSnapshot();
});

test('handles request action', () => {
  const beforeState = {
    isLoading: false,
  };
  const afterState = usersReducer(beforeState, {type: 'USERS_REQUEST'});

  expect(afterState).toEqual({
    isLoading: true,
  });
});

test('handles success action', () => {
  const beforeState = {
    isLoading: true,
  };
  const afterState = usersReducer(beforeState, {
    type: 'USERS_GET_SUCCESS',
    payload: {result: [1, 2, 3]},
  });

  expect(afterState).toEqual({
    isLoading: false,
    result: [1, 2, 3],
  });
});

test('handles failure action', () => {
  const beforeState = {
    isLoading: true,
    error: null,
  };
  const afterState = usersReducer(beforeState, {
    type: 'USERS_FAILURE',
    error: true,
    payload: new Error('fail'),
  });

  expect(afterState).toMatchSnapshot();
});
