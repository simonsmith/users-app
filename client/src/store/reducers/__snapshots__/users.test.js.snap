// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles failure action 1`] = `
Object {
  "error": [Error: fail],
  "isLoading": false,
}
`;

exports[`should handle no state being passed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "result": Array [],
}
`;
